frontend/src/components/account/inventory/Inventory.js

```diff
@@ -12,16 +12,148 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+import { Link } from "react-router-dom";
+import "./Inventory.css";
+import ComponentWithHeader from "../../header/ComponentWithHeader";
 
-class Inventory extends Component {
+class Inventory extends ComponentWithHeader {
+  constructor(properties) {
+    super(properties);
+    this.state = {
+      inventory: [],
+      edit: false,
+    };
+
+    this.addItem = this.addItem.bind(this);
+    this.deleteItem = this.deleteItem.bind(this);
+  }
+  
+  setEdit() {
+    this.getInventory();
+    const previousEdit = this.state.edit;
+    this.setState({edit: !previousEdit});
+  }
+  editLabel(){
+      if (this.state.edit){
+        return "Cancel";
+      } else {
+        return "Edit"
+      }
+  }
+  editCssLabel(){
+      if (this.state.edit){
+        return "edit-inventory";
+      } else {
+        return "hidden-inventory"
+      }
+  }
+  addItem(event) {
+    const value = this._inputElement.value;
+    if (value === "") {
+      event.preventDefault();
+      return;
+    }
+    var newItem = {
+      text: value.charAt(0).toUpperCase() + value.slice(1).toLowerCase(),
+      key: Date.now(),
+    };
+    var isDuplicate = false;
+    this.setState((prevState) => {
+      prevState.inventory.forEach(function (item, index, object) {
+        if (item.text === newItem.text) {
+          isDuplicate = true;
+          window.alert("Duplicate Ingredient");
+        }
+      });
+      if (isDuplicate) {
+        return {
+          inventory: prevState.inventory,
+        };
+      } else {
+        return {
+          inventory: [newItem].concat(prevState.inventory),
+        };
+      }
+    });
+    this._inputElement.value = "";
+    event.preventDefault();
+  }
+
+  deleteItem(key) {
+    var filteredinventory = this.state.inventory.filter(function (item) {
+      return item.key !== key;
+    });
+
+    this.setState({
+      inventory: filteredinventory,
+    });
+  }
+  componentDidMount() {
+    this.getInventory();
+  }
     
-    render() {
-        return(
-            <body>
-            <h1>Inventory</h1>
-            </body>
-        )
+  renderContent() {
+    const inventory = this.state.inventory.map((item) => item.text);
+    return(
+      <div>
+        <div className="centered-container">
+          <h1 className="account-page-title">Inventory</h1>
+          <Button className="edit-button" onClick={()=>this.setEdit()}>{this.editLabel()}</Button>
+          <Form className={this.editCssLabel()} onSubmit={this.addItem}>
+            <Form.Row>
+              <div className="input">
+                <Form.Control
+                  ref={(a) => (this._inputElement = a)}
+                  type="text"
+                  placeholder="New Ingredient"
+                />
+              </div>
+              <div>
+                <Button type="submit">Add</Button>
+              </div>
+            </Form.Row>
+          </Form>
+        <InventoryItems editState={this.editCssLabel()} entries={this.state.inventory}
```

Comment:
Add a line that describes what the user sees, something similar to "Here are the items that you always have in your kitchen" (doesn't have to be this exact message)